nifi-nar-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/PutBoxFile.java

```diff
@@ -0,0 +1,366 @@
+                    .name("success")
+                    .description("Files that have been successfully written to Box are transferred to this relationship.")
+                    .build();
+
+    public static final Relationship REL_FAILURE =
+            new Relationship.Builder()
+                    .name("failure")
+                    .description("Files that could not be written to Box for some reason are transferred to this relationship.")
+                    .build();
+
+    public static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(asList(
+            REL_SUCCESS,
+            REL_FAILURE
+    )));
+
+    public static final int CONFLICT_RESPONSE_CODE = 409;
+
+    private volatile BoxAPIConnection boxAPIConnection;
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final String filename = context.getProperty(FILE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+        String folderId = context.getProperty(FOLDER_ID).evaluateAttributeExpressions(flowFile).getValue();
+        final String subfolderName = context.getProperty(SUBFOLDER_NAME).evaluateAttributeExpressions(flowFile).getValue();
+        final boolean createFolder = context.getProperty(CREATE_SUBFOLDER).asBoolean();
+        BoxFile.Info uploadedFile = null;
+
+        final long startNanos = System.nanoTime();
+
+        String fullPath = null;
+
+        try {
+            final long size = flowFile.getSize();
+
+            folderId = subfolderName != null ? getOrCreateParentSubfolder(subfolderName, folderId, createFolder).getID() : folderId;
+            final BoxFolder parentFolder = getFolder(folderId);
+            fullPath = BoxFileUtils.getPath(parentFolder.getInfo());
+
+            final long chunkUploadThreshold = context.getProperty(CHUNKED_UPLOAD_THRESHOLD)
+                    .asDataSize(DataUnit.B)
+                    .longValue();
+
+            final String conflictResolution = context.getProperty(CONFLICT_RESOLUTION).getValue();
+
+            try (InputStream rawIn = session.read(flowFile)){
+
+                boolean isNewVersionUpload = false;
+
+                if (REPLACE_RESOLUTION.equals(conflictResolution)) {
+                    final Optional<BoxFile.Info> alreadyUploadedFile = getFileByName(filename, parentFolder);
+
+                    if (alreadyUploadedFile.isPresent()) {
+                        BoxFile existingBoxFile = new BoxFile(boxAPIConnection, alreadyUploadedFile.get().getID());
+                        existingBoxFile.uploadNewVersion(rawIn);
```

Comment:
`uploadedFile` should be assigned here too. Otherwise the FF attributes will not be populated and no provenance event will be sent for replaced files.